server/src/main/java/org/opensearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -114,7 +115,11 @@ public QueryPhaseResultConsumer(
 
         SearchSourceBuilder source = request.source();
         this.hasTopDocs = source == null || source.size() != 0;
-        this.hasAggs = source != null && source.aggregations() != null;
+        if (FeatureFlags.isEnabled(FeatureFlags.PROTOBUF)) {
```

Comment:
@VachaShah I think I understand why we do this (not implemented) but from the feature perspective it looks like cut off in the middle, may be we could make the decision upfront (we know if request includes aggs) and always use the `native` protocol instead?

Reply:
I think we cannot make the decision upfront since transport is separate from the rest, by the time the request/response comes to transport, we don't really have a view of if it contains aggs or not. But I think we can remove this, since we haven't implemented it. I was actually doing this for some tests, but its not needed. Let me fix this in the next commit.